src/scanner.c

```diff
@@ -417,18 +444,30 @@ static inline bool scan_close_bracket2(TSLexer* lexer, Stack* stack) {
 }
 
 static bool scan(TSLexer* lexer, Stack* stack, const bool* valid_symbols) {
+  if (valid_symbols[ERROR_SENTINEL]) {
+    // Decline to handle when in "error recovery" mode. When a syntax error occurs,
+    // tree-sitter calls the external scanner with all `valid_symbols` marked as valid.
+    return false;
+  }
+
+  if (valid_symbols[START]) {
+    // The `START` symbol is only valid at the very beginning of a file before we
+    // have seen any tokens. We emit this zero width symbol to force the `program`
+    // node to open at position `(0, 0)`, regardless of how much leading whitespace
+    // (including both `' '` and `\r`) there may be before our first "real" token.
+    // This ensures the AST spans the entire file, which consumers of it rely on (#151).
+    lexer->result_symbol = START;
+    return true;
+  }
```

Comment:
Main change 2 of 2

If we see `valid_symbols[START]`, then the ONLY way that it possible is if we are at the start of the file before we've seen any tokens. Emit a zero width `START` external which forces the `program` node to open at `(0, 0)`.

Reply:
I really expected that tree-sitter would help us out here and force `program` to start at `(0, 0)`, but it does not and it does not seem to be something other languages care about

For example, if you put this in the tree-sitter playground and add some leading newlines before `def`
https://tree-sitter.github.io/tree-sitter/playground

```python


def x():
  1
```

then you should see that `module` does not start at `(0, 0)`